tests/modules/js-and-css/optimize-embeds/optimize-oembeds-test.php

```diff
@@ -0,0 +1,101 @@
+				'<iframe loading="lazy" src="https://www.youtube.com/embed/123" width="560" height="315" frameborder="0"></iframe>',
+			),
+			// An iframe and a script.
+			array(
+				'<iframe src="https://www.youtube.com/embed/123" width="560" height="315" frameborder="0"></iframe><script src="https://www.youtube.com/embed/123"></script>',
+				'<iframe loading="lazy" src="https://www.youtube.com/embed/123" width="560" height="315" frameborder="0"></iframe><script data-lazy-embed-src="https://www.youtube.com/embed/123" ></script>',
+			),
+			// An iframe that is already lazy loaded.
+			array(
+				'<iframe src="https://www.youtube.com/embed/123" width="560" height="315" frameborder="0" loading="lazy"></iframe>',
+				'<iframe src="https://www.youtube.com/embed/123" width="560" height="315" frameborder="0" loading="lazy"></iframe>',
+			),
+			// An iframe and an inline script.
+			array(
+				'<iframe src="https://www.youtube.com/embed/123" width="560" height="315" frameborder="0"></iframe><script>console.log("Hello, World!");</script>',
+				'<iframe loading="lazy" src="https://www.youtube.com/embed/123" width="560" height="315" frameborder="0"></iframe><script>console.log("Hello, World!");</script>',
+			),
+			// An instagram embed.
+			array(
+				'<blockquote class="instagram-media" data-instgrm-captioned data-instgrm-permalink="https://www.instagram.com/p/123/" data-instgrm-version="13"></blockquote>',
+				'<blockquote class="instagram-media" data-instgrm-captioned data-instgrm-permalink="https://www.instagram.com/p/123/" data-instgrm-version="13"></blockquote>',
+			),
+			// Vimeo embed.
+			array(
+				'<iframe loading="lazy" title="Expedia - Plates (DC)" src="https://player.vimeo.com/video/806311047?dnt=1&amp;app_id=122963" width="500" height="281" frameborder="0" allow="autoplay; fullscreen; picture-in-picture"></iframe>',
+				'<iframe loading="lazy" title="Expedia - Plates (DC)" src="https://player.vimeo.com/video/806311047?dnt=1&amp;app_id=122963" width="500" height="281" frameborder="0" allow="autoplay; fullscreen; picture-in-picture"></iframe>',
+			),
+			// Twitter embed.
+			array(
+				'<blockquote class="twitter-tweet" data-width="500" data-dnt="true"><p lang="en" dir="ltr"Feedback <a href="https://t.co/anGk6gWkbc">https://t.co/anGk6gWkbc</a></p>&mdash; <a href="https://twitter.com/ChromiumDev/status/1636796541368139777?ref_src=twsrc%5Etfw">March 17, 2023</a></blockquote><script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>',
+				'<blockquote class="twitter-tweet" data-width="500" data-dnt="true"><p lang="en" dir="ltr"Feedback <a href="https://t.co/anGk6gWkbc">https://t.co/anGk6gWkbc</a></p>&mdash; <a href="https://twitter.com/ChromiumDev/status/1636796541368139777?ref_src=twsrc%5Etfw">March 17, 2023</a></blockquote><script data-lazy-embed-src="https://platform.twitter.com/widgets.js" async  charset="utf-8"></script>',
```

Comment:
I just had a thought. What if instead of converting `script[src]' to `script[data-lazy-embed-src]`, it instead set the `type` attribute of `text/x-lazy-loaded-javascript`? Clearly out of scope for this PR, but just thought of this alternative. It probably doesn't make any difference, oh except actually wouldn't it be better in that it would prevent the browser from treating this `script` as an inline script? While the browser hopefully is smart enough to skip parsing an inline script that has no contents, in the worst case it could. That would be bad because inline scripts are blocking. If the `src` is left-as is and the `type` is used to mark such lazy-loaded scripts instead, this would be avoided. This is also what Partytown does to mark its scripts.

Reply:
interesting! I haven't seen this approach but it makes sense that it would work.  feel free to propose on another PR.

Reply:
Here you go! https://github.com/WordPress/performance/pull/983